pkg/gclient/metadata/testing/client.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2020 Google LLC.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"github.com/google/knative-gcp/pkg/gclient/metadata"
+)
+
+var (
+	clusterNameAttr = "cluster-name"
+	FakeClusterName = "fake-cluster-name"
+	FakeProjectID   = "fake-project-id"
+)
+
+type testMetadataClient struct {
+}
+
+func NewTestClient() metadata.Client {
```

Comment:
Ideally this function could take some error mocking parameter(s) like https://github.com/google/knative-gcp/blob/f780beb7298acadbe75785fa2b49795a4f4066df/pkg/gclient/scheduler/testing/client.go#L49-L56
Then it would be possible to test error cases in `annotations_test.go`. That can be a followup if you prefer.